aiida_common_workflows/workflows/relax/generator.py

```diff
@@ -68,28 +85,49 @@ def get_builder(
         threshold_forces=None,
         threshold_stress=None,
         previous_workchain=None,
+        is_insulator=False,
+        spin=SpinType.NONE,
+        initial_magnetization='auto',
         **kwargs
     ):
         """Return a process builder for the corresponding workchain class with inputs set according to the protocol.
 
-        :param structure: the structure to be relaxed
-        :param calc_engines: ...
-        :param protocol: the protocol to use when determining the workchain inputs
-        :param relaxation_type: the type of relaxation to perform, instance of `RelaxType`
+        :param structure: the structure to be relaxed.
+        :param calc_engines: a dictionary containing the computational resources for the relaxation.
+        :param protocol: the protocol to use when determining the workchain inputs.
+        :param relaxation_type: the type of relaxation to perform, instance of `RelaxType`.
         :param threshold_forces: target threshold for the forces in eV/Å.
         :param threshold_stress: target threshold for the stress in eV/Å^3.
         :param previous_workchain: a <Code>RelaxWorkChain node.
+        :param is_insulator: a bool to activate insulator options (by default metal calcs are assumed.
+        :param spin: the spin state of the calculation, instance of `RelaxType`.
+        :param initial_magnetization: a list with the spin polarization for each site.
         :param kwargs: any inputs that are specific to the plugin.
         :return: a `aiida.engine.processes.ProcessBuilder` instance ready to be submitted.
         """
-        if previous_workchain:
+        if previous_workchain is not None:
             try:
                 prev_wc_class = previous_workchain.process_class
                 if not prev_wc_class == self.process_class:
                     raise ValueError('The "previous_workchain" must be a node of {}'.format(self.process_class))
             except AttributeError:
                 raise ValueError('The "previous_workchain" must be a node of {}'.format(self.process_class))
 
+        if relaxation_type not in self._relaxation_types:
+            raise ValueError('relaxation type `{}` is not supported'.format(relaxation_type))
+
+        if is_insulator not in [False, True, None]:
+            raise ValueError('The argument `is_insulator` accepts only `False`, `True` or `None`')
+
+        if spin not in self._spin_types:
+            raise ValueError('spin type `{}` is not supported'.format(spin))
+
+        if initial_magnetization != 'auto':
+            if not isinstance(initial_magnetization, list):
```

Comment:
probably also want to check that the list is flat and contains only floats/integers

Reply:
Ok